src/metrics/prometheus.go

```diff
@@ -0,0 +1,42 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/expfmt"
+	"gopkg.in/op/go-logging.v1"
+)
+
+var log = logging.MustGetLogger("metrics")
+
+// RemoteMetrics is a type for maintaining please remote metrics
+type RemoteMetrics struct {
+	downloadErrorCounter prometheus.Counter
+}
+
+// NewRemoteMetrics initialises the remote metrics
+func NewRemoteMetrics() *RemoteMetrics {
+	downloadErrorCounter := prometheus.NewCounter(prometheus.CounterOpts{
+		// Note: this can be called multiple times and won't affect the gateway's counter value.
+		Name: "tree_digest_download_eof_error",
+		Help: "Number of times the Unexpected EOF error has been seen during a tree digest download",
+	})
+
+	return &RemoteMetrics{
+		downloadErrorCounter: downloadErrorCounter,
+	}
+}
+
+// DownloadErrorCounterInc increments the 'tree digest download error' counter
+func (r *RemoteMetrics) DownloadErrorCounterInc(gatewayURL string) {
```

Comment:
Might make more sense to store the gateway url as a string in the struct and pass it in to NewRemoteMetrics()

Reply:
Yep that is neater (I've also made it just a metrics singleton now, which can be used for any future metrics), thanks Jon :)